### PR TITLE
[checkbox-ce-oem] add iio sensors test (New)

### PR DESCRIPTION
**Pull request by stanley31huang**
## Description
Implement new test cases for industrial I/O sensor

## Resolved issues
N/A

## Documentation
Refer to the kernel document about the Industrial I/O sensor
https://www.kernel.org/doc/html/v5.4/driver-api/iio/core.html#industrial-i-o-devices

## Tests
Tested on Murcia project, please refer the test results from C3 website.
https://certification.canonical.com/hardware/202312-33237/submission/359230/

List-bootstrapped logs
```
stanley@stanley-ThinkPad-T490:~$ checkbox-cli list-bootstrapped com.canonical.contrib::ce-oem-iio-sensors-full
Using sideloaded provider: checkbox-provider-ce-oem, version 0.1 from /var/tmp/checkbox-providers/checkbox-provider-ce-oem
Skipped file: /usr/share/checkbox-provider-base/units/camera/README.rst
Skipped file: /usr/share/checkbox-provider-base/units/snapd/README.md
Skipped file: /usr/share/checkbox-provider-base/units/stress/suspend_cycles_reboot.md
Using sideloaded provider: checkbox-provider-ce-oem, version 0.1 from /var/tmp/checkbox-providers/checkbox-provider-ce-oem
com.canonical.plainbox::manifest
com.canonical.contrib::ce-oem-iio-sensors/check-pressure-0
com.canonical.certification::rtc
com.canonical.certification::sleep
com.canonical.certification::suspend/suspend_advanced_auto
com.canonical.contrib::after-suspend-ce-oem-iio-sensors/check-pressure-0
com.canonical.contrib::ce-oem-iio-sensors/check-accelerometer-1
com.canonical.contrib::after-suspend-ce-oem-iio-sensors/check-accelerometer-1
com.canonical.contrib::ce-oem-iio-sensors/check-humidityrelative-2
com.canonical.contrib::after-suspend-ce-oem-iio-sensors/check-humidityrelative-2
```

### CODE REVIEW

**Review by baconYao — APPROVED** [review]
LGTM

File: contrib/checkbox-provider-ce-oem/units/iio-sensors/test-plan.pxu [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+include:
+
+id: ce-oem-iio-sensors-automated
+unit: test plan
+_name: Industrial I/O sensor auto tests
+_description: Automated tests for Industrial I/O sensors
+bootstrap_include:
+    ce-oem-iio-sensors/resource
+include:
+    ce-oem-iio-sensors/check-.*-.*
```
Comment (rickwu666666) [inline contrib/checkbox-provider-ce-oem/units/iio-sensors/test-plan.pxu:25]: This regular expression is almost the same as "ce-oem-iio-sensors/check-.*". What's the reason for using this "-.*-.*"?

**Review by rickwu666666 — COMMENTED** [review]
A minor question. Please see my in-line comment.
Otherwise LGTM

File: contrib/checkbox-provider-ce-oem/units/iio-sensors/test-plan.pxu (hunk 1 above)
Comment (stanley31huang) [reply contrib/checkbox-provider-ce-oem/units/iio-sensors/test-plan.pxu]: Fixed this issue by include the template-id in the test plan

**Review by rickwu666666 — APPROVED** [review]
LGTM

**Review by baconYao — APPROVED** [review]
LGTM

File: contrib/checkbox-provider-ce-oem/units/test-plan-ce-oem.pxu [hunk 2]
```diff
@@ -76,6 +77,7 @@ nested_part:
     ce-oem-ethernet-tcp-automated
     com.canonical.certification::eeprom-automated
     com.canonical.certification::rtc-automated
+    ce-oem-iio-sensors-automated
```
Comment (LiaoU3) [inline contrib/checkbox-provider-ce-oem/units/test-plan-ce-oem.pxu:80]: I think it should be `before-suspend`?

**Review by LiaoU3 — CHANGES_REQUESTED** [review]
Please see my comment.

File: contrib/checkbox-provider-ce-oem/units/test-plan-ce-oem.pxu (hunk 2 above)
Comment (stanley31huang) [reply contrib/checkbox-provider-ce-oem/units/test-plan-ce-oem.pxu:79]: you're right, thanks for point out this issue.

File: contrib/checkbox-provider-ce-oem/bin/iio_sensor_test.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+    Raises:
+        FileNotFoundError: the sysfs of sensor not exists
+
+    Returns:
+        iio_node: the node of the industrial I/O sensor. (Path object)
+    """
+    iio_node = Path(path)
+    if not iio_node.exists():
+        raise FileNotFoundError("{} file not exists".format(str(iio_node)))
```
[the remark this answers is not in the document]
Comment (stanley31huang) [reply contrib/checkbox-provider-ce-oem/bin/iio_sensor_test.py]: thanks for point out this issue.

[the remark this answers is not in the document]
Comment (stanley31huang) [reply contrib/checkbox-provider-ce-oem/bin/iio_sensor_test.py]: fixed

File: contrib/checkbox-provider-ce-oem/units/test-plan-ce-oem.pxu (hunk 2 above)
Comment (LiaoU3) [reply contrib/checkbox-provider-ce-oem/units/test-plan-ce-oem.pxu:79]:
```suggestion
    before-suspend-ce-oem-iio-sensors-automated
```

File: contrib/checkbox-provider-ce-oem/bin/iio_sensor_test.py [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+        ValueError: the reading of sensor is not expected format
+    """
+    iio_node = _check_node(IIO_PATH + index)
+    readings = []
+
+    for sub_node in pressure_nodes:
+        tmp_node = iio_node.joinpath(sub_node)
+        _check_node(tmp_node)
+        value = tmp_node.read_text().strip("\n")
+        print("the value of {} node is {}".format(tmp_node, value))
```
Comment (LiaoU3) [inline contrib/checkbox-provider-ce-oem/bin/iio_sensor_test.py:104]:
```suggestion
        print("The value of {} node is {}".format(tmp_node, value))
```

**Review by LiaoU3 — CHANGES_REQUESTED** [review]
Please see my comments.

**Review by LiaoU3 — APPROVED** [review]
LGTM

**Review by rickwu666666 — APPROVED** [review]
LGTM

**Pull request merged**